src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -14,171 +14,179 @@ import AlertDialogBox from '../../../../components/AlertDialogBox';
-    this.state.applicants.forEach((profile) =>
-      this.offCampusProgramInfo.set(profile.AD_Username, ''),
-    );
-    this.editDialogText = '';
-    this.snackbarText = '';
-    this.snackbarSeverity = '';
-    this.saveButtonAlertTimeout = null;
-  }
-
-  componentDidMount() {
-    this.loadProfile();
-    this.loadSavedApplication();
-  }
+const InstructionsCard = () => (
+  <Card>
+    <CardHeader title="Apartment Application Instructions" className="card-header" />
+    <CardContent>
+      <Typography variant="body1">Placeholder Text</Typography>
+      <Typography variant="body1">
+        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde
+        suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam
+        dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
+      </Typography>
+    </CardContent>
+  </Card>
+);
+
+const StudentApplication = ({ userProfile }) => {
+  const [loading, setLoading] = useState(true);
+  const [saving, setSaving] = useState(false);
+
+  const [applicationID, setApplicationID] = useState(-1); // Default value of -1 indicate to backend that the application ID number is not yet known
+  const [editorUsername, setEditorUsername] = useState(null); // The username of the application editor
+  const [applicants, setApplicants] = useState([]);
+  const [preferredHalls, setPreferredHalls] = useState([{ HallRank: 1, HallName: '' }]);
+  // const [offCampusProgramInfo, setOffCampusProgramInfo] = useState(new Map());
+
+  const [applicationCardsOpen, setApplicationCardsOpen] = useState(false);
+  const [newEditorProfile, setNewEditorProfile] = useState(null); // Stores the StudentProfileInfo of the new editor before the user confirms the change
+  const [changeEditorDialogOpen, setChangeEditorDialogOpen] = useState(false);
+  const [submitDialogOpen, setSubmitDialogOpen] = useState(false); // Use this for submitting app (later feature)
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState('');
+  const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
+
+  // const peopleSearchRef = useRef();
+
+  let offCampusProgramInfo = new Map();
+  applicants.forEach((profile) => offCampusProgramInfo.set(profile.AD_Username, ''));
 
   /**
-   * Loads the user's profile info only once (at start)
+   * Attempt to load an existing application from the database if one exists
    */
-  async loadProfile() {
-    this.setState({ loading: true });
-    try {
-      let applicants = this.state.applicants;
-      applicants.push(this.props.userProfile);
-      this.setState({ applicants });
-      this.setState({ loading: false });
-    } catch (error) {
-      // Do Nothing
-    }
-  }
+  useEffect(() => {
+    loadSavedApplication();
+
+    // eslint-disable-next-line
+  }, []);
```

Comment:
EsLint is trying to warn you about a legitimate bug, in this case. The loadSavedApplication function actually gets redefined on every render. This is not really impactful for performance but it does mean that the useEffect would have to rerun every render because the function keeps changing. The way you have it now, it will probably work 99% of the time but is not idiomatic React. The bug is that, if the component stayed mounted but the userProfile prop changed, this useEffect wouldn't reload so you wouldn't load the new application.

The simplest way to fix this is to define loadSavedApplication within the useEffect. Then, you can properly declare it's single dependency userProfile and the useEffect will run at the needed times (first render and whenever userProfile changes). There's another way to fix this using useCallback, but that is significantly more complex and is only necessary when the function (i.e. loadSavedApplication in this case) needs to be called elsewhere in the component outside the useEffect. Since you aren't doing that, I think the simpler fix is the way to go for you.

Reply:
This function actually will need to be called elsewhere in the file for several features that are coming soon, so I will have to research that other solution today

Reply:
I think I may have over stated it's complexity. You essentially just need to wrap the function declaration in the useCallback hook. The second argument of useCallback is the dependencies of the function (in this case, just userProfile I think). You also need to define the callback before you use it anywhere, so it would have to come before the useEffect call.

 i.e. `const loadSavedApplication = useCallback(async () => {...}, [userProfile]);`

Reply:
Thanks, I was confused when the EsLint used the phrase "wrap the function declaration in ....", but I didn't have a chance to look it up at the time.
After seeing an example like that it makes perfect sense